* The build will fail if liblockfile isn't selected for movemail.
  Patch: require-movemail-use-liblockfile.diff
  Author: Rob Browning <rlb@defaultvalue.org>
  Added-by: Rob Browning <rlb@defaultvalue.org>

  This makes sure the Debian Emacs won't accidentally be built with
  the wrong locking strategy.  To disable this check, comment out
  require-movemail-use-liblockfile.diff in debian/patches/series.

--- a/lib-src/movemail.c
+++ b/lib-src/movemail.c
@@ -166,6 +166,10 @@
 /* Nonzero means this is name of a lock file to delete on fatal error.  */
 char *delete_lockname;
 
+#ifndef MAIL_USE_MAILLOCK
+#error "Debian requires that mail locking be handled by liblockfile."
+#endif /* ndef MAIL_USE_MAILLOCK */
+
 int
 main (argc, argv)
      int argc;
--- a/src/s/bsd-common.h
+++ b/src/s/bsd-common.h
@@ -66,7 +66,11 @@
    The alternative is that a lock file named
    /usr/spool/mail/$USER.lock.  */
 
+/* conditional copied from gnu-linux.h */
+#if !((defined (HAVE_LIBMAIL) || defined (HAVE_LIBLOCKFILE)) && \
+      defined (HAVE_MAILLOCK_H))
 #define MAIL_USE_FLOCK
+#endif
 
 /* Define CLASH_DETECTION if you want lock files to be written
    so that Emacs can tell instantly when you try to modify
